File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
           if(nums.length==0) return 0;
        if(nums.length==1) return nums[0];
        int maxSum=nums[0];
        
        int usm=0;
        for (int i = 0; i < nums.length; i++) {
            usm+=nums[i];
            if(usm>maxSum){maxSum=usm;}
            if(usm<0)usm=0;
        }



        return maxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        localMax,globalmax=0,0
        
        globalmax=nums[0]
        
        for i in range(len(nums)):
            localMax= max(localMax+nums[i],nums[i])
            globalmax= max(globalmax,localMax)
        return (globalmax)